crates/re_ui/examples/re_ui_example.rs

```diff
@@ -600,3 +647,751 @@ impl egui_tiles::Behavior<Tab> for MyTileTreeBehavior {
+            // TODO(emilk/egui#3841): it would be nice to have response.decidedly_dragged()
+            if response.dragged() {
+                // Here, we support dragging a single item at a time, so we set the selection to the dragged item
+                // if/when we're dragging it proper.
+                if ui.input(|i| i.pointer.is_decidedly_dragging()) {
+                    self.selected_items.clear();
+                    self.selected_items.insert(*idx);
+                }
+            }
+
+            //
+            // Detect end-of-drag situation and prepare the swap command.
+            //
+
+            // TODO(emilk/egui#3841): very tempting to use `response.dragged()` here, but it
+            // doesn't work. By the time `i.pointer.any_released()` is true, `response.dragged()`
+            // is false. So both condition never happen at the same time.
+            if ui.memory(|mem| mem.is_being_dragged(response.id)) {
+                source_item_pos = Some(i);
+            }
+
+            // TODO(emilk/egui#3841): this feels like a common enough pattern that is should deserve its own API.
+            let anything_being_decidedly_dragged = ui.memory(|mem| mem.is_anything_being_dragged())
+                && ui.input(|i| i.pointer.is_decidedly_dragging());
+            if anything_being_decidedly_dragged {
+                let (top, bottom) = response.rect.split_top_bottom_at_fraction(0.5);
+
+                let (insert_y, target) = if ui.rect_contains_pointer(top) {
+                    (Some(top.top()), Some(i))
+                } else if ui.rect_contains_pointer(bottom) {
+                    (Some(bottom.bottom()), Some(i + 1))
+                } else {
+                    (None, None)
+                };
+
+                if let Some(insert_y) = insert_y {
+                    ui.painter().hline(
+                        ui.cursor().x_range(),
+                        insert_y,
+                        (2.0, egui::Color32::WHITE),
+                    );
+
+                    // TODO(emilk/egui#3841): it would be nice to have a drag specific API for that
+                    if ui.input(|i| i.pointer.any_released()) {
+                        target_item_pos = target;
+                    }
+                }
+            }
+        }
+
+        //
+        // Handle the swap command (if any)
+        //
+
+        if let (Some(source), Some(target)) = (source_item_pos, target_item_pos) {
+            if source != target {
+                let item = self.items.remove(source);
+
+                if source < target {
+                    self.items.insert(target - 1, item);
+                } else {
+                    self.items.insert(target, item);
+                }
+            }
+        }
+    }
+}
+
+// ==============================================================================
+// HIERARCHICAL DRAG AND DROP DEMO
```

Comment:
Put it in its own `mod` and?or file?

Reply:
I've paid the rightward drift and used a mod. I'm unsure how to make multi-file examples.